src/read.c

```diff
@@ -2802,10 +2866,46 @@ static avifBool avifParseTrackBox(avifDecoderData * data, uint64_t rawOffset, co
             AVIF_CHECK(avifParseMediaBox(track, rawOffset + avifROStreamOffset(&s), avifROStreamCurrent(&s), header.size, data->diag));
         } else if (!memcmp(header.type, "tref", 4)) {
             AVIF_CHECK(avifTrackReferenceBox(track, avifROStreamCurrent(&s), header.size, data->diag));
+        } else if (!memcmp(header.type, "edts", 4)) {
+            if (edtsBoxSeen) {
+                avifDiagnosticsPrintf(data->diag, "More than one [edts] Box was found.");
+                return AVIF_FALSE;
+            }
+            AVIF_CHECK(avifParseEditsBox(track, avifROStreamCurrent(&s), header.size, data->diag));
+            edtsBoxSeen = AVIF_TRUE;
         }
 
         AVIF_CHECK(avifROStreamSkip(&s, header.size));
     }
+    if (!edtsBoxSeen) {
+        track->repetitionCount = AVIF_REPETITION_COUNT_UNKNOWN;
+    } else if (track->isRepeating) {
+        if (track->trackDuration == AVIF_UNKNOWN_DURATION64) {
+            // If isRepeating is true and track duration is unknown, then set the repetition count to infinite (Section 9.6.1 of
+            // ISO/IEC 23008-12 Part 12).
+            track->repetitionCount = AVIF_REPETITION_COUNT_INFINITE;
+        } else {
+            // Section 9.6.1. of ISO/IEC 23008-12 Part 12: 1, the entire edit list is repeated a sufficient number of times to
+            // equal the track duration.
+            //
+            // Since libavif uses repetitionCount (which is 0-based), we subtract the value by 1 to derive the number of
+            // repetitions.
+            assert(track->segmentDuration != 0);
+            const uint64_t repetitionCount =
+                (track->trackDuration / track->segmentDuration) + (track->trackDuration % track->segmentDuration != 0) - 1;
```

Comment:
Nit: The `(a + b - 1) / b` expression is commonly used to calculate the ceiling of the `a / b` quotient (with normal, real-number division). Do you think `(a / b) + (a % b != 0)` is easier to understand?

Reply:
I believe both forms are okay. I generally prefer the `%` to circumvent the problem of `(a + b - 1)` intermediate addition potentially overflowing.

Reply:
If `track->trackDuration` is 0, the new expression will evaluate to 0 + 0 - 1. We should defend against this case or return an error.

Reply:
Ah yes.

I have added a check for trackDuration == 0.

Note that i am specifically doing it here and not where trackDuration is read. Since we only use the value in this case, we don't have to punish files which had accidentally set it to 0 without the editlist with older tools (those files would have parsed successfully so far because we never read the trackDuration value until this PR).